exec/java-exec/src/main/resources/rest/static/js/graph.js

```diff
@@ -93,41 +93,82 @@ $(window).on('load',(function () {
     }
 
     // parse the short physical plan into a dagreeD3 structure
-    function parseplan (planstring) {
+    function parseplan (planstring, implicitSrcMap) {
         var g = new dagreD3.Digraph();
-        var ps = $.map(planstring.trim().split("\n"), function (s) {
-            return [/^([0-9-]+)( *)([a-zA-Z]*)/.exec(s).slice(1)];
-        });
+        //Produce 2D array (3 x M): [[0:majorMinor] [1:] [2:opName]]
+        let opPlanArray = planstring.trim().split("\n");
+        var operatorRegex = new RegExp("^([0-9-]+)( *)([a-zA-Z]*)");
+        //Regex to capture source operator 
+        var srcOpRegex = new RegExp("srcOp=[0-9-]+");
+        var opTuple = $.map(opPlanArray, 
+            function (lineStr) { 
+              //Tokenize via Regex
+              let opToken = operatorRegex.exec(lineStr).slice(1);
+              //Extract Implicit Source via Regex
+              let implicitSrc = srcOpRegex.exec(lineStr);
+              let srcOp = null;
+              let tgtOp = opToken[0];
+              if (implicitSrc != null) {
+                srcOp = implicitSrc.toString().split("=")[1];
+                implicitSrcMap[tgtOp]=srcOp;
+              }
+              return [opToken];
+            });
+
 
-        // nodes
-        for (var i = 0; i < ps.length; i++) {
-            g.addNode(ps[i][0], {
-                label: ps[i][2] + " " + ps[i][0],
-                fragment: parseInt(ps[i][0].split("-")[0])
+        // parse, build & inject nodes
+        for (var i = 0; i < opTuple.length; i++) {
+            let majorMinor = opTuple[i][0];
+            let majorId = parseInt(majorMinor.split("-")[0]);
+            let opName = opTuple[i][2];
+            g.addNode(majorMinor, {
+                label: opName + " " + majorMinor,
+                fragment: majorId
             });
         }
 
-        // edges
-        var st = [ps[0]];
-        for (var i = 1; i < ps.length; i++) {
-            var top = st.pop();
-            while (top[1].length >= ps[i][1].length)
-                top = st.pop();
+        // Define edges by traversing graph from root node (Screen)
```

Comment:
I would recommend adding a comment here that `"the indentation value in opTuple which is opTuple[1] represents the relative level of each operator in tree w.r.t root operator"`

Reply:
Done